src/linear_algebra/basis.lean

```diff
@@ -739,6 +739,101 @@ end
 
 end span
 
+lemma eq_zero_of_smul_eq_zero {G : Type*} [hG : group G] [distrib_mul_action G R]
+  (g : G) {r : R} (h : g • r = 0) : r = 0 :=
+begin
+  suffices : g⁻¹ • g • r = 0,
+  { rw [← this, ← smul_assoc, smul_eq_mul, mul_left_inv, one_smul] },
+  rw [h, smul_zero],
+end
```

Comment:
The version of this in my PR is `smul_eq_zero_iff_eq`. I think the proof works just fine here:
```lean
-- TODO: remove when #7438 is merged
theorem smul_eq_zero_iff_eq (a : α) {x : β} : a • x = 0 ↔ x = 0 :=
⟨λ h, by rw [← inv_smul_smul a x, h, smul_zero], λ h, h.symm ▸ smul_zero _⟩
```

I made a thread about whether we should wait for #7438 here: https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.237438.20actions.20on.20and.20by.20units/near/239288252

Reply:
I put #7438 in the merge queue